src/framework-ondata.js

```diff
@@ -1,18 +1,301 @@
+ *
+ * @param port the error port, which may or may not be attached to something down stream
+ * @param lastLm last recorded state lm
+ */
+function handleError( vni, port, lastLm  ) { 
+
+    if ( haveError( vni ) )  {
+
+        var state = vni.errorState();
+
+        if ( (! handleOutput( port, lastLm, state )) && state.data )  { 
+            // State was not sent on to an attached port, and we do have error data
+            // so go ahead and log it for debugging/support use.
+            if ( state.data.stack ) { 
+                console.error( state.data.stack );
+            } else { 
+                console.error( state.data );
+            }
+        }
+    }
+}
+
+/**
+ * Returns true if the state is an initial vni state that has not got any
+ * values in it yet.
+ */
+function isInitState( state ) { 
+    return ( _.isUndefined( state.data ) && _.isUndefined( state.lm ) && _.isUndefined( state.error ) );
+}
+
+/**
+ * Checks if state is new or not.  If it is new, it sends it down stream to any attached 
+ * nodes; if there are no attached nodes, the state data will be logged.
+ *
+ * @param port an output port, which may or may not be attached to something down stream
+ * @param lastLm last recorded state lm
+ * @param state an error or output state to be sent down stream or logged
+ *
+ * @return true if the data was sent to an attached port on a down stream node;
+ *         false if the data was not sent on
+ */
+function handleOutput( port, lastLm, state ) {
+
+    // Do we have a new state or an error flag set? 
+    if ( lastLm !== state.lm || state.error ) { 
+
+        // Got any edges out of this port? 
+        if (port.listAttached().length) {
+            port.send( state );
+            port.disconnect();
+            return true;
+        }  
+    } 
+
+    return false;
+}
+
+/**
+ * Set the output state error flag to true if there is a current error, false if not.
+ * 
+ * @param vni 
+ * @param errorFlag (optional) if not undefined, the errorFlag value will be set
+ */
+function setOutputErrorFlag( vni, errorFlag ) {
+
+    var outputState  = vni.outputState();
+    if ( _.isUndefined( errorFlag ) ) { 
+        outputState.error = ! _.isUndefined( vni.errorState().data );
+    } else { 
+        outputState.error = errorFlag;
+    }
+
+    vni.outputState( outputState );
+}
+
+/** 
+ * Checks if the state was changed since the last state snapshot & if so, 
+ * updates the LM to ensure the change propagates.
+ *
+ * @param stateFacade a getter/setterfunction for the current state
+ * @param lastState a saved clone of the last state 
+ *
+ * @return returns true if state was updated, false if not.
+ */
+function stateChange( stateFacade, lastState ) { 
+
+    var currentState = stateFacade();
+
+    // If got new data or an error state
+    var lastData = (lastState &&  _.isObject( lastState )  ) ? lastState.data : lastState;
+    if ( currentState.data  !== lastData ) { 
+
+        currentState.lm = createLm();
```

Comment:
Only generate a new LM for error object -- not for output object.  Better to do it outside of this function, so that this function does not have a side effect.